akka-javasdk/src/main/java/akka/javasdk/agent/MemoryProvider.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+/**
+ * Interface for configuring memory management in agent systems.
+ * <p>
+ * MemoryProvider defines how conversation history is stored and retrieved during agent interactions.
+ * It offers several implementation strategies:
+ * <ul>
+ *   <li>Configuration-based memory management via {@link FromConfig}</li>
+ *   <li>Limited window memory management via {@link LimitedWindowMemoryProvider}</li>
+ *   <li>Custom memory implementation via {@link CustomMemoryProvider}</li>
+ * </ul>
+ */
+public sealed interface MemoryProvider {
+
+  /**
+   * Creates a memory provider based on configuration settings.
+   *
+   * @param configPath Path to the configuration. If empty, uses the default path "akka.javasdk.agent.memory"
+   * @return A configuration-based model provider
+   */
+  static ModelProvider fromConfig(String configPath) {
+    return new ModelProvider.FromConfig(configPath);
+  }
+
+  /**
+   * Configuration-based memory provider that reads settings from the specified path.
+   */
+  record FromConfig(String configPath) implements MemoryProvider {}
+
+  /**
+   * Creates a limited window memory provider with default settings.
+   * <p>
+   * The default settings include:
+   * <ul>
+   *   <li>No limit on total length in bytes</li>
```

Comment:
As mentioned during the standup, maybe we should in fact have a hard limit on bytes. Maybe then this setting will be obsolete?